### PR TITLE
Various improvements :-)

### PR DESCRIPTION
**Pull request by kristoffer-zliide**
Details are in the commit comments.

### CODE REVIEW

File: android/src/main/java/de/simolus3/fluttie/FluttiePlugin.java [hunk 1]
```diff
@@ -12,31 +13,26 @@
...
-import io.flutter.plugin.common.EventChannel;
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
 import io.flutter.plugin.common.MethodChannel.Result;
 import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.PluginRegistry.Registrar;
 import io.flutter.view.TextureRegistry;
 
-public class FluttiePlugin implements MethodCallHandler, EventChannel.StreamHandler, Application.ActivityLifecycleCallbacks {
+public class FluttiePlugin implements MethodCallHandler, Application.ActivityLifecycleCallbacks {
```
Comment (simolus3) [inline android/src/main/java/de/simolus3/fluttie/FluttiePlugin.java:28]: Thanks for migrating the usage of an event stream to the regular async method call API here, makes it much easier to understand what's going on :+1:

File: README.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ In order to display animations in Flutter, you will have the plugin load
 a composition first. The plugin will parse the composition file so
 that it can quickly display the animation later on.
 ```dart
-var instance = new Fluttie();
+var instance = Fluttie();
```
Comment (simolus3) [inline README.md:49]: Thanks for this clean-up

File: android/src/main/java/de/simolus3/fluttie/FluttiePlugin.java [hunk 3]
```diff
@@ -90,27 +84,22 @@ private FluttieAnimation getManagedAnimation(MethodCall call) {
...
+	RenderingThreads getRenderingThreads() {
 		return renderingThreads;
 	}
 
 	@Override
 	public void onMethodCall(MethodCall call, Result result) {
 		switch (call.method) {
 			case "isAvailable":
-				result.success(true);
+				result.success(Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP);
```
Comment (simolus3) [inline android/src/main/java/de/simolus3/fluttie/FluttiePlugin.java:95]: I see why we have do this, but AFAIK the cause of this crash has already been fixed in the Flutter engine and it no longer occurs on the `master` branch of flutter, right?
I'm wondering whether this is so critical that we really have to disable the plugin on pre-Lollipop... But then again we can re-enable it in a subsequent version. What do you think?

**Review by simolus3 — APPROVED** [review]
Thanks cleaning up the README and the Dart code. Also, thanks a lot for getting rid of that hacky stream code and for using the async method calling api directly. I originally used it because I thought one was supposed to reply to incoming calls asap, but as it turns out it can be done much simpler. :+1: 
I just have the question on disabling the plugin on pre-lollipop devices. Do you think it's necessary to do that? I guess the fix from flutter will be merged into `beta` soon right?

File: android/src/main/java/de/simolus3/fluttie/FluttiePlugin.java (hunk 3 above)
Comment (kristoffer-zliide) [reply android/src/main/java/de/simolus3/fluttie/FluttiePlugin.java:95]: I thought about that too :-)

I am completely unable to predict when they choose to update the beta branch. It's been five weeks now and I think this summer they went more than eight weeks without updating it. Also, as the situation is now, developers may install default Flutter, add the latest version of your fine plugin, build their app, and publish it in the store. All without a chance of knowing that they just published an app that crashes on pre-Lollipop phones. I'd rather not have that and then having to roll back this commit when they decide to release a new beta.

Comment (simolus3) [reply android/src/main/java/de/simolus3/fluttie/FluttiePlugin.java:95]: That is a very good point! I'll merge this PR as it is & create a new version then, as we can still revert that part later.

**Comment by kristoffer-zliide** [comment]
So, they released flutter 0.10.2 and now it no longer crashes. But it doesn't look good either :-(

screenshot_2018-11-08-16-54-39

**Pull request merged**